assessments/projects/contour/self-assessment.md

```diff
@@ -0,0 +1,206 @@
+to perform an internal analysis of the project's security. It is not intended to provide 
+a security audit of Contour, or function as an independent assessment or attestation of Contour's security health.
+
+This document serves to provide Contour users with an initial understanding of
+Contour's security, where to find existing security documentation, Contour plans for
+security, and general overview of Contour security practices, both for development of
+Contour as well as security of Contour.
+
+This document provides the CNCF TAG-Security with an initial understanding of Contour
+to assist in a joint-assessment, necessary for projects under incubation.  Taken
+together, this document and the joint-assessment serve as a cornerstone for if and when
+Contour seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+* Critical
+  * xDS server – Functions as a control plane for Envoy. The xDS server knows all the secrets that Contour knows, so Contour requires a mutual TLS authentication between Contour and Envoy such that another cluster cannot simply connect to the xDS server and collect all the known secrets.
+* Security Relevant
+  * Federal Information Processing Standard (FIPS) 140-2 – FIPS describes United States government approved secuirty requirements for cryptographib modules. Software validated by an accredited Crpytographic Module Validation Program (CVMP) labortory is suitable for use in applications for US governmental departments or industries subjected to US Federnal regulations. Contour is not FIPS validated but can be built and configured to the standards FIPS 140-2 establishes.
+
+## Project compliance
+
+* Compliance
+  * TBD after discussion with project maintainers
+
+## Secure development practices
+
+* Development Pipeline - Contour utilizes GitHub issues, labels, and code reviews for secure open source development.
+  <br>
+  > A detailed description of secure development practices can be found at https://projectcontour.io/resources/how-we-work/.
+
+  When GitHub issues are created and being worked on, the contributor assigned to the issue is responsible for giving regular status reports on the issue. Additionally, contributors should not work on issues that they are not assigned to, horde multiple issues, or assign issues to others without communicating first. Each of these practices ensure that issues are well tracked and understood as to avoid a decline in productivity and security.
+  GitHub issues should also be labeled, as appropriate labels make it much easier for an issue to be assessed at any given time. 
+
+  There are also code reviews, where a reviewer can review and make comments on a contributor’s code. 
+  Before fixing any bug, a test must be written to show that the bug was fixed. Similarly, tests must be written for new features to prevent a future developer from accidentally breaking it down the line.
+  <br>
+* Communication Channels
```

Comment:
Can you please include the inline references to each of the communication methods specified below?

Reply:
references have been included.